invenio_vocabularies/datastreams/readers.py

```diff
@@ -416,6 +416,8 @@ def __init__(self, origin, query, mode="r", *args, **kwargs):
         """
         self._origin = origin
         self._query = query
+        self._user_agent = kwargs.get("user_agent", "")
```

Comment:
```suggestion
        self._client_params = client_params or {}
```

minor: I would convert this to a `client_params` dictionary-style value (and add it as a param in `SPARQLReader.__init__`), so that we can pass arbitrary arguments in the future. This is also to allow just not passing `user_agent` at all, so that the default value from the `SPARQLWrapper` class.

Reply:
In principle I like this idea! See comment below for my issues with implementation